src/App/Fossa/Analyze.hs

```diff
@@ -59,10 +60,10 @@ data ScanDestination
   | OutputStdout
   deriving (Generic)
  
-analyzeMain :: Severity -> ScanDestination -> Maybe Text -> Maybe Text -> IO ()
-analyzeMain logSeverity destination name revision = do
-  basedir <- getCurrentDir
-  withLogger logSeverity $ analyze basedir destination name revision
+analyzeMain :: Severity -> FilePath -> ScanDestination -> Maybe Text -> Maybe Text -> IO ()
+analyzeMain logSeverity basedir destination name revision = do
+  basedir' <- validateDir basedir
```

Comment:
We should probably validate these _before_ we call `analyzeMain`, and maybe use a `newtype` to enforce validation.